Feature/viewer

kimhyeri opened the pull request:
- searBar animation 버그 해결 했습니다.

- filterView 선택한 label이 지워지는 버그 해결했습니다. 
  - (category연결은 이제 진행할 예정입니다.)

- tableView cell을 터치할 경우 아티클 뷰어 push는 한번만 실행되도록 했습니다. 

- 한 pickerView를 공유하기 때문에 type을 지정해 case애 맞는 처리를 진행했습니다.

- 앱에서 모두 똑같은 navigationBar를 설정하기 때문에 setupNavigationBar는 모든 UIViewController에서 사용 가능하도록 extension시켰습니다.

kimhyeri committed:
Fear : add loading view

kimhyeri committed:
Refactor : Extract setup navigation func
#93 네비게이션 설정을 UIViewController extension으로 뺐습니다.
searchKeyword는 Optional으로 주었습니다.

kimhyeri committed:
Refactor : Code convention
#93 if let 을 guard let으로 변경했습니다.

kimhyeri committed:
Refactor :  Code convention
#93 change method name

kimhyeri committed:
Merge remote-tracking branch 'refs/remotes/origin/develop'

kimhyeri committed:
Refactor : Update info plist file
statusBar 설정하는 코드 변경했습니다.

kimhyeri committed:
Refactor : Code convention
#93 override touchesBegan func

kimhyeri committed:
Refactor :  Code convention
#93 font 변경, 소문자 대문자로 변경

kimhyeri committed:
Refactor :  Extract filter type enum
#92 #93 버그 해결 및 리펙토링

kimhyeri committed:
Add : SearchBar bug fixed
#92 Solved searchBar bug
Override didRotate func

kimhyeri committed:
Refactor : Fixed didSelectRowAt func bug
Changed font type
Changed method name
#92 #93

kimhyeri committed:
Feat : Connect webView scrap button
웹뷰에서 스크랩 할 수 있도록 IBAction 연결했습니다.

kimhyeri committed:
Refactor : Change if let statement
#93 if let 구문을 guard let 구문으로 변경했습니다.

yagom commented on tree/tree/Controller/Search/ArticleViewer/ArticleWebViewController.swift:
```diff
@@ -67,6 +67,10 @@ class ArticleWebViewController: UIViewController {
         self.navigationController?.popViewController(animated: true)
     }
+    
+    @IBAction func scrapButtonDidTap(_ sender: UIBarButtonItem) {
```
기능이 구현되지 않아서 비어있는거죠?

kimhyeri replied:
네 이 부분은 현태님이 구현할 자리라 비워놨습니다! 😆

kimhyeri committed:
Refactor : Change project setting
Set key color
Change file directory

kimhyeri committed:
Refactor : Add categoryUri param
#92 사용자가 선택한 카테고리에 맞는 검색 결과 호출 가능하도록 변경

kimhyeri committed:
Refactor : setup statusBarStyle
Code convention
#92 #93

godpp commented on tree/tree/Controller/Search/SearchFilterViewController.swift:
```diff
@@ -74,13 +76,27 @@ class SearchFilterViewController: UIViewController {
-                makeHidden(isPresented: selectViewIsPresented) : 
+            switch selectPickViewer.pickerType {
+            case .category:
+                makeHidden(isPresented: selectViewIsPresented)
```
makeHidden(by : selectViewIsPresented) 외부 인자 이름을 사용하는게 낫지 않을까요?!?! 🤔

kimhyeri replied:
네 외부인자 이름을 사용하는것도 좋겠네요.
makeHidden(by isPresented: Bool) 로 변경하겠습니다.

godpp commented on tree/tree/Controller/Search/SearchViewController.swift:
```diff
@@ -360,15 +372,15 @@ extension SearchViewController: UISearchBarDelegate {
+            !isLoading else { return }
+        self.navigationItem.title = getSearchKeyword
+        isLoading.toggle()
+        searchKeyword = getSearchKeyword
```
```swift
self.searchKeyword = searchKeyword
```
이건 어떠신가요? 단순히 searchBar.text를 getSearchKeyword라고 부르는건 개인적으로는 조금 부자연스러운것 같습니다!

kimhyeri replied:
넵 수정하겠습니당

godpp commented on tree/tree/Model/DataManager/ScrapService.swift:
```diff
@@ -65,7 +65,7 @@ public enum ArticleCategory: CaseIterable{
         let baseCategory: NSString = NSString(string: "\(self)")
         let firstCharactor = baseCategory.character(at: 0)
-        guard let unicode = UnicodeScalar(firstCharactor - 32) as? UnicodeScalar else {
+        guard let unicode = UnicodeScalar(firstCharactor - 32) else {
```
불필요한 함수로 보입니다! Capitailize는
```swift
"abc".capitalized
```
이걸로 가능합니다!

kimhyeri replied:
네 이 부분은 지우도록 하겠습니다!

kimhyeri replied:
네 삭제하겠습니다.

godpp reviewed (commented):
수고하셨습니다!

코멘트 확인 부탁드려요!

kimhyeri committed:
Refactor : Code convention
#93

godpp reviewed (approved):
수고하셨습니다! 👍

kim-howard commented on tree/tree/Controller/Search/SearchViewController.swift:
```diff
@@ -125,38 +136,44 @@ class SearchViewController: UIViewController {
             let language = searchFilter["language"], 
-            let sort = searchFilter["sort"] 
+            let sort = searchFilter["sort"],
+            let category = searchFilter["category"]
```
저는 이런 keyword , language 같이 직접 키를 작성하는 부분은 enum으로 작성해서 string 을 넣는것이 더 좋다고 생각합니다! 👍

kimhyeri replied:
okay ! 좋은 생각입니다

kim-howard commented on tree/tree/Controller/Search/SearchViewController.swift:
```diff
@@ -125,38 +136,44 @@ class SearchViewController: UIViewController {
                 language: language, 
-                sort: sort
+                sort: sort,
+                category: "dmoz/\(category)"
```
왜 굳이 dmoz를 붙여 놓으신건가요?! 물론 같이 api결과로 넘어오는것은 알고있지만 category 자체에는 그 값만 들어가는 것이 더 좋다고 생각합니다.

kimhyeri replied:
dmoz가 파라미터로 같이 들어가야해서 dmoz를 string으로 함께 넣는 방법을 선택했습니다.

kim-howard commented on tree/tree/Controller/Search/SearchViewController.swift:
```diff
@@ -125,38 +136,44 @@ class SearchViewController: UIViewController {
+        category: String
     ) {
         articles = nil
+        guard let searchBarText = searchKeyword else { return }
```
굳이 이름을 다르게 하신 이유가 있나요?! 만약에 큰 이유가 없다면 seachdBarText 혹은 searchKeyword 로 하나만 쓰는게 낫지 않을까요?!

kimhyeri replied:
이 부분은 이전에 변경했습니다!
감사합니당

kim-howard reviewed (approved):
확인 부탁드려요 ! ^^

kimhyeri committed:
Refactor : Code convention
#93 searchFilter enum 추가했습니다.
